crates/astria-proto/src/native/sequencer/ics20_withdrawal.rs

```diff
@@ -0,0 +1,234 @@
+    },
+    IdentifierError,
+};
+use penumbra_proto::penumbra::core::component::ibc::v1alpha1::FungibleTokenPacketData;
+
+use super::{
+    asset::{
+        IbcAsset,
+        IbcAssetError,
+    },
+    v1alpha1::IncorrectAddressLength,
+};
+use crate::{
+    generated::{
+        primitive::v1::Uint128,
+        sequencer::v1alpha1 as raw,
+    },
+    native::sequencer::v1alpha1::Address,
+};
+
+#[derive(Debug, Clone)]
+pub struct Ics20Withdrawal {
+    // a transparent value consisting of an amount and a denom.
+    amount: u128,
+    denom: IbcAsset,
+    // the address on the destination chain to send the transfer to.
+    destination_chain_address: String,
+    // an Astria address to use to return funds from this withdrawal
+    // in the case it fails.
+    return_address: Address,
+    // the height (on Astria) at which this transfer expires.
+    timeout_height: IbcHeight,
+    // the timestamp at which this transfer expires.
+    timeout_time: u64,
+    // the source channel used for the withdrawal.
+    source_channel: ChannelId,
+}
+
+impl Ics20Withdrawal {
+    #[must_use]
+    pub fn amount(&self) -> u128 {
+        self.amount
+    }
+
+    #[must_use]
+    pub fn denom(&self) -> &IbcAsset {
+        &self.denom
+    }
+
+    #[must_use]
+    pub fn destination_chain_address(&self) -> &str {
+        &self.destination_chain_address
+    }
+
+    #[must_use]
+    pub fn return_address(&self) -> &Address {
+        &self.return_address
+    }
+
+    #[must_use]
+    pub fn timeout_height(&self) -> &IbcHeight {
+        &self.timeout_height
+    }
+
+    #[must_use]
+    pub fn timeout_time(&self) -> u64 {
+        self.timeout_time
+    }
+
+    #[must_use]
+    pub fn source_channel(&self) -> &ChannelId {
+        &self.source_channel
+    }
+
+    #[must_use]
+    pub fn to_fungible_token_packet_data(&self) -> FungibleTokenPacketData {
+        FungibleTokenPacketData {
+            amount: self.amount.to_string(),
+            denom: self.denom.to_string(),
+            sender: self.return_address.to_string(),
+            receiver: self.destination_chain_address.clone(),
+        }
+    }
+
+    #[must_use]
+    pub fn to_raw(&self) -> raw::Ics20Withdrawal {
+        raw::Ics20Withdrawal {
+            amount: Some(self.amount.into()),
+            denom: self.denom.to_string(),
+            destination_chain_address: self.destination_chain_address.clone(),
+            return_address: self.return_address.0.to_vec(),
+            timeout_height: Some(self.timeout_height.into()),
+            timeout_time: self.timeout_time,
+            source_channel: self.source_channel.to_string(),
+        }
+    }
+
+    #[must_use]
+    pub fn into_raw(self) -> raw::Ics20Withdrawal {
+        raw::Ics20Withdrawal {
+            amount: Some(self.amount.into()),
+            denom: self.denom.to_string(),
+            destination_chain_address: self.destination_chain_address,
+            return_address: self.return_address.0.to_vec(),
```

Comment:
```suggestion
            return_address: self.return_address.to_vec(),
```
`Address` has a `to_vec` method